File: arrays/convert1DArrayInto2DArray.py
# https://leetcode.com/problems/convert-1d-array-into-2d-array/

from typing import List

class Solution:
    # Runtime: 1014 ms, faster than 96.23% of Python3 online submissions for Convert 1D Array Into 2D Array.
    # Memory Usage: 21.3 MB, less than 95.27% of Python3 online submissions for Convert 1D Array Into 2D Array.
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        
        if len(original) != m*n : return []
        
        return [original[i: i+n] for i in range (0,len(original),n)]
    
    
    

File: arrays/convert1DArrayInto2DArray.js
// Runtime: 395 ms, faster than 60.12% of JavaScript online submissions for Convert 1D Array Into 2D Array.
// Memory Usage: 62.4 MB, less than 90.19% of JavaScript online submissions for Convert 1D Array Into 2D Array.

/**
 * @param {number[]} original
 * @param {number} m
 * @param {number} n
 * @return {number[][]}
 */
const construct2DArray = (original, m, n) => {
    let row = [];
    if(original.length !== m * n){
        return [];
    }
    for (let i = 0; i < original.length; i += n){
        row.push(original.slice(i,i+n));
    }
    return row;
    
};
module.exports = construct2DArray;